nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -305,16 +309,28 @@ def _wrap_model_for_O2(self):
 
         args = mcore_args if is_mcore_model else nemo_args
         # Model wrapper to convert both model and inputs to half precision
-        if isinstance(self.model, list):
-            converted_model = []
-            for module in self.model:
-                args['module'] = module
-                converted_model.append(Float16Wrapper(**args))
-            self.model = converted_model
+        if hasattr(self, "enc_dec_model"):
```

Comment:
can you de-duplicate if and else. maybe 
if hasattr(self, "enc_dec_model") model = self.enc_dec_model else model = self.model
and then act on model

Reply:
Thanks for the note Yu Yao.

I actually have tried your method before (combining self.model and self.enc_dec_model then act on self.model only).
But it introduces problems down the line, because many existing T5-related scripts act on self.enc_dec_model, hence sometimes there's a misunderstanding/mismatch between self.model and self.enc_dec_model, which causes incorrect training.
In the end, I find it might be best to keep it consistent by working on self.enc_dec_model across code files.

Anyhow, I still simplify using your suggestion `(self.enc_dec_model if hasattr(self, "enc_dec_model") else self.model)` to reduce if-else duplicate.